From 5de87f6a01f9067e0dc94092419ec01e16aebe86 Mon Sep 17 00:00:00 2001
From: Joel A Fernandes <joelagnel@ti.com>
Date: Thu, 25 Oct 2012 00:02:27 -0500
Subject: [PATCH 1/9] arm/dts: add am33xx mcasp1 dt node

Add AM33xx McASP1 DT node. Device tree Data is split into SoC specific
and Board specific.

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/boot/dts/am335x-boneblack.dts |   14 ++++++++++++++
 arch/arm/boot/dts/am33xx.dtsi       |    9 +++++++++
 2 files changed, 23 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-boneblack.dts b/arch/arm/boot/dts/am335x-boneblack.dts
index 87257ac..a97ee36 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -30,3 +30,17 @@
 	ti,non-removable;
 	status = "okay";
 };
+
+
+&mcasp0 {
+	op-mode = <0>;          /* MCASP_IIS_MODE */
+	tdm-slots = <2>;
+	num-serializer = <16>;
+	serial-dir = <  /* 0: INACTIVE, 1: TX, 2: RX */
+                      0 0 1 2
+                      0 0 0 0
+                      0 0 0 0
+                      0 0 0 0 >;
+	tx-num-evt = <1>;
+	rx-num-evt = <1>;
+};
diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index 8bddc9c..15e04d4 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -258,6 +258,15 @@
 			status = "disabled";
 		};
 
+		mcasp0: mcasp@4803C000 {
+			compatible = "ti,omap2-mcasp-audio";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			ti,hwmods = "mcasp1";
+			reg = <0x4803C000 0x2000>;
+			interrupts = <82 83>;
+		};
+
 		mmc1: mmc@48060000 {
 			compatible = "ti,omap3-hsmmc";
 			ti,hwmods = "mmc1";
-- 
1.7.7.6

